### DISCUSSION TITLE
The HTTP server runs on localhost:8080 instead of http://0.0.0.0:8080

### DISCUSSION THREAD

**Discussion by DablSi** [aio-libs/aiohttp#4979] [Q&A] [answered]
🐞 **Describe the bug**
Both tutorial and terminal says:
> ======== Running on http://0.0.0.0:8080 ========
> (Press CTRL+C to quit)

But if you open this exact URL,  both Firefox and Edge browsers will say "Connection attempt failed".
Nonetheless I've accidentally opened http://localhost:8080. And that's exactly where the server was running.

💡 **To Reproduce**
1. I am using the last version of PyCharm Community Edition on Windows 10
2. I was trying to reproduce some code from the [polls example](http://demos.aiohttp.org/en/latest/tutorial.html#aiohttp-demos-polls-getting-started)
```python
from aiohttp import web

app = web.Application()
web.run_app(app)
```

💡 **Expected behavior**
If you open http://0.0.0.0:8080, both Firefox and Edge browsers will say "Connection attempt failed".

📋 **Your version of the Python**
Python 3.8

📋 **Your version of the aiohttp/yarl/multidict distributions**
aiohttp 3.6.2
multidict 4.7.6
yarl 1.6.0

**Comment by webknjaz** [comment, marked as answer]
[`0.0.0.0`] is a catch-all wildcard for binding to all the interfaces available on the machine as opposed to listening to only one specific address. Strictly speaking, it does not have any semantical meaning in terms of routing traffic to a specific location. So when the client says "I want to send some bytes to 0.0.0.0", technically this doesn't mean anything. While saying "send the traffic to 127.0.0.1" (or "localhost" for that matter) should work because the networking subsystem can look that up in the routing table and will know where to send your queries over the network.

Using [`0.0.0.0`] only makes sense on the server-side in terms of binding to machine-local network interfaces. Some operating systems (namely GNU/Linux based distros), the client software stack intelligently substitutes [`0.0.0.0`] under the hood, while in others (like Windows, in your case) this doesn't happen and, as a result, the network stack has no clue where to send your queries to. Ref: https://superuser.com/a/536198/182499

What aiohttp log outputs is the ephemeral bind-all address that it listens to. But it needs to be accessed by the concrete address your network interfaces have. That may be either the localhost loopback interface (127.0.0.1) or some other address your computer got assigned in your LAN. Remember: the loopback only works for connections from within the same host (operating system to be precise) while other addresses may be accessible from the LAN given that they are routable.

[`0.0.0.0`]: https://en.wikipedia.org/wiki/0.0.0.0

Ref:
* https://www.howtogeek.com/225487/what-is-the-difference-between-127.0.0.1-and-0.0.0.0/

**Comment by azizali20990** [comment]
ufw allow 8080

**Comment by sekhmilan84-lab** [comment]
K vil k

**Comment by sekhmilan84-lab** [comment]
Jfbj
